apps/token/src/routes/governance/hooks/use-vote-information.ts

```diff
@@ -1,162 +1,185 @@
-    NetworkParams.governance_proposal_freeform_requiredParticipation,
-  ]);
-
-  if (!params) {
-    return {
-      requiredMajority: new BigNumber(1),
-      requiredParticipation: new BigNumber(1),
-    };
-  }
-
-  switch (proposal.terms.change.__typename) {
-    case 'UpdateMarket':
-      return {
-        requiredMajority:
-          params.governance_proposal_updateMarket_requiredMajority,
-        requiredParticipation: new BigNumber(
-          params.governance_proposal_updateMarket_requiredParticipation
-        ),
-      };
-    case 'UpdateNetworkParameter':
-      return {
-        requiredMajority:
-          params.governance_proposal_updateNetParam_requiredMajority,
-        requiredParticipation: new BigNumber(
-          params.governance_proposal_updateNetParam_requiredParticipation
-        ),
-      };
-    case 'NewAsset':
-      return {
-        requiredMajority: params.governance_proposal_asset_requiredMajority,
-        requiredParticipation: new BigNumber(
-          params.governance_proposal_asset_requiredParticipation
-        ),
-      };
-    case 'NewMarket':
-      return {
-        requiredMajority: params.governance_proposal_market_requiredMajority,
-        requiredParticipation: new BigNumber(
-          params.governance_proposal_market_requiredParticipation
-        ),
-      };
-    case 'NewFreeform':
-      return {
-        requiredMajority: params.governance_proposal_freeform_requiredMajority,
-        requiredParticipation: new BigNumber(
-          params.governance_proposal_freeform_requiredParticipation
-        ),
-      };
-    default:
-      throw new Error('Unknown proposal type');
-  }
-};
+import { useProposalNetworkParams } from './use-proposal-network-params';
+import type { Proposal_proposal } from '../proposal/__generated__/Proposal';
 
 export const useVoteInformation = ({
   proposal,
 }: {
-  proposal: ProposalFields;
+  proposal: Proposal_proposal;
 }) => {
   const {
     appState: { totalSupply },
   } = useAppState();
 
-  const { requiredMajority, requiredParticipation } = useProposalNetworkParams({
+  const {
+    requiredMajority,
+    requiredParticipation,
+    requiredMajorityLP,
+    requiredParticipationLP,
+  } = useProposalNetworkParams({
     proposal,
   });
 
-  const requiredMajorityPercentage = React.useMemo(
+  const requiredMajorityPercentage = useMemo(
     () =>
       requiredMajority
         ? new BigNumber(requiredMajority).times(100)
         : new BigNumber(100),
     [requiredMajority]
   );
 
-  const noTokens = React.useMemo(() => {
+  const requiredMajorityLPPercentage = useMemo(
+    () =>
+      requiredMajorityLP
+        ? new BigNumber(requiredMajorityLP).times(100)
+        : new BigNumber(100),
+    [requiredMajorityLP]
+  );
+
+  const noTokens = useMemo(() => {
     return new BigNumber(proposal.votes.no.totalTokens);
   }, [proposal.votes.no.totalTokens]);
 
-  const yesTokens = React.useMemo(() => {
+  const noELSWeight = useMemo(() => {
```

Comment:
suggestion (non blocking): Took me a bit too long to remember ELS is equity like share, can we rename to `noEquityLikeShareWeight`?